tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -0,0 +1,1647 @@
+
+        const str op_rstr_literal = str{get_cat<CharType>()} + get_dog<CharType>();
+        assert(equalRanges(op_rstr_literal, "kittendog"sv));
+
+        const str op_rstr_char = str{get_cat<CharType>()} + CharType{'!'};
+        assert(equalRanges(op_rstr_char, "kitten!"sv));
+
+        const str op_str_rstr = first + str{get_dog<CharType>()};
+        assert(equalRanges(op_str_rstr, "kittendog"sv));
+
+        const str op_literal_rst = get_cat<CharType>() + str{get_dog<CharType>()};
+        assert(equalRanges(op_rstr_literal, "kittendog"sv));
+
+        const str op_char_rstr = CharType{'!'} + str{get_dog<CharType>()};
+        assert(equalRanges(op_char_rstr, "!dog"sv));
+    }
+
+    { // comparison
+        str first(get_view_input<CharType>());
+        str second(get_view_input<CharType>());
+        str third{get_cat<CharType>()};
+
+        const bool eq_str_str = first == second;
+        assert(eq_str_str);
+
+        const bool ne_str_str = first != third;
+        assert(ne_str_str);
+
+        const bool less_str_str = first < third;
+        assert(less_str_str);
+
+        const bool less_eq_str_str = first <= third;
+        assert(less_eq_str_str);
+
+        const bool greater_str_str = first > third;
+        assert(!greater_str_str);
+
+        const bool greater_eq_str_str = first >= third;
+        assert(!greater_eq_str_str);
+
+        const bool eq_str_literal = first == get_view_input<CharType>();
+        assert(eq_str_literal);
+
+        const bool ne_str_literal = first != get_cat<CharType>();
+        assert(ne_str_literal);
+
+        const bool less_str_literal = first < get_cat<CharType>();
+        assert(less_str_literal);
+
+        const bool less_eq_str_literal = first <= get_cat<CharType>();
+        assert(less_eq_str_literal);
+
+        const bool greater_str_literal = first > get_cat<CharType>();
+        assert(!greater_str_literal);
+
+        const bool greater_eq_str_literal = first >= get_cat<CharType>();
+        assert(!greater_eq_str_literal);
+
+        const bool eq_literal_str = get_view_input<CharType>() == second;
+        assert(eq_literal_str);
+
+        const bool ne_literal_str = get_view_input<CharType>() != third;
+        assert(ne_literal_str);
+
+        const bool less_literal_str = get_view_input<CharType>() < third;
+        assert(less_literal_str);
+
+        const bool less_eq_literal_str = get_view_input<CharType>() <= third;
+        assert(less_eq_literal_str);
+
+        const bool greater_literal_str = get_view_input<CharType>() > third;
+        assert(!greater_literal_str);
+
+        const bool greater_eq_literal_str = get_view_input<CharType>() >= third;
+        assert(!greater_eq_literal_str);
+    }
+#endif // defined(MSVC_INTERNAL_TESTING) || defined(__EDG__) || _ITERATOR_DEBUG_LEVEL != 0
+#endif // __EDG__
+    return true;
+}
+
+template <class CharType = char>
+_CONSTEXPR20_CONTAINER bool test_iterators() {

```

Comment:
I believe we are missing test coverage of `operator+(iter, iter)`, `op=`, and `op->`.